dm/ui/src/components/CreateTaskConfig/MigrateRule.tsx

```diff
@@ -0,0 +1,258 @@
+                      <Form.Item
+                        label={t('source')}
+                        name={[field.key, 'source', 'source_name']}
+                        {...itemLayout}
+                      >
+                        <Select
+                          onChange={val => setCurrentSource(val as string)}
+                        >
+                          {data?.data.map(source => (
+                            <Select.Option
+                              key={source.source_name}
+                              value={source.source_name}
+                            >
+                              {source.source_name}
+                            </Select.Option>
+                          ))}
+                        </Select>
+                      </Form.Item>
+                      <Form.Item
+                        label={t('database')}
+                        name={[field.key, 'source', 'schema']}
+                        {...itemLayout}
+                      >
+                        <Input
+                          placeholder="*"
+                          addonAfter={
+                            <Cascader
+                              options={cascaderOptions}
+                              loadData={loadCascaderData}
+                              multiple
+                              maxTagCount="responsive"
+                              value={cascaderValue}
+                              onChange={(value: any) => {
+                                setCascaderValue(value)
+                                const dbPattern = createPattern([
+                                  ...new Set(
+                                    value.map((item: string[]) => item[0])
+                                  ),
+                                ] as string[])
+                                const tablePattern = createPattern([
+                                  ...new Set(
+                                    value
+                                      .map((item: string[]) => item[1])
+                                      .filter(Boolean)
+                                  ),
+                                ] as string[])
+                                const newData = [
+                                  ...form.getFieldValue('table_migrate_rule'),
+                                ]
+                                newData[field.key] = {
+                                  ...newData[field.key],
+                                  source: {
+                                    ...newData[field.key].source,
+                                    schema: dbPattern,
+                                    table: tablePattern,
+                                  },
+                                }
+                                form.setFieldsValue({
+                                  table_migrate_rule: newData,
+                                })
+                              }}
```

Comment:
Maybe better to move to a method?

Reply:
sure, I'll move it out later in another pr